mlir/lib/Analysis/Presburger/Matrix.cpp

```diff
@@ -29,6 +29,24 @@ Matrix<T>::Matrix(unsigned rows, unsigned columns, unsigned reservedRows,
   data.reserve(std::max(nRows, reservedRows) * nReservedColumns);
 }
 
+template <typename T>
+bool Matrix<T>::operator==(const Matrix<T> &m) const {
+  if (nRows != m.getNumRows())
+    return false;
+  if (nColumns != m.getNumColumns())
+    return false;
+
+  for (unsigned i = 0; i < nRows; i++) {
+    for (unsigned j = 0; j < nColumns; j++) {
+      if (at(i, j) == m.at(i, j))
```

Comment:
if !=
  return false

Reply:
you can then skip the braces too.

Reply:
oh, i thought that avoiding excessive nesting was the first priority